libs/x-content/src/main/java/org/opensearch/common/xcontent/MediaType.java

```diff
@@ -62,4 +62,6 @@ public interface MediaType {
     default String typeWithSubtype() {
         return type() + "/" + subtype();
     }
+
+    XContent xContent();
```

Comment:
Just came to mind, do you think we could introduce `XContentMediaType` to break `XContentType` and `MediaType`:

```
interface XContentMediaType extends MediaType {
     XContent xContent()
}
```

And than we would have `XContentType implements XContentMediaType` ?
Sorry about that, didn't see that before.

Reply:
Are you then wanting to keep the XContent base classes in the x-content library instead of refactoring like #5902? If so, we would still need to tease out those classes because so much of the gaggregation and field mappers framework depend on XContent base classes. 

If not, then this doesn't buy us anything until we refactor the entire `:server` codebase to be less dependent on `XContent` and more `MediaType` generic

Reply:
Yeah, I wanted to keep `MediaType` clean, right now it won't buy us anything but later the `XContent` won't be an obstacle going to "more MediaType generic" stage.

Reply:
Let's add that abstraction in a follow up PR then.

Reply:
I can take it, btw with `XContentMediaType` we could backport it to `2.x` as well, cuz `MediaType` won't be change and `XContentType` as well :-)

Reply:
Sorry. I didn't clarify well. We should add it in a follow up PR that makes `:server` less dependent on `XContentType`. If we do this now it will block refactoring the `Strings` class to core which is a core class needed to refactor a LOT of other base classes out of `:server`.